src/test/java/com/hpe/adm/octane/ideplugins/integrationtests/IntegrationTestBase.java

```diff
@@ -13,30 +13,55 @@
 
 package com.hpe.adm.octane.ideplugins.integrationtests;
 
+import com.google.api.client.json.Json;
 import com.google.inject.Guice;
 import com.google.inject.Inject;
 import com.google.inject.Injector;
+import com.hpe.adm.nga.sdk.Octane;
+import com.hpe.adm.nga.sdk.authentication.Authentication;
+import com.hpe.adm.nga.sdk.authentication.SimpleUserAuthentication;
+import com.hpe.adm.nga.sdk.model.EntityModel;
+import com.hpe.adm.nga.sdk.network.OctaneHttpClient;
+import com.hpe.adm.nga.sdk.network.OctaneHttpRequest;
+import com.hpe.adm.nga.sdk.network.OctaneHttpResponse;
+import com.hpe.adm.nga.sdk.network.OctaneRequest;
+import com.hpe.adm.nga.sdk.network.google.GoogleHttpClient;
 import com.hpe.adm.octane.ideplugins.integrationtests.util.EntityGenerator;
 import com.hpe.adm.octane.ideplugins.integrationtests.util.PropertyUtil;
+import com.hpe.adm.octane.ideplugins.integrationtests.util.WorkSpace;
+import com.hpe.adm.octane.ideplugins.services.EntityService;
 import com.hpe.adm.octane.ideplugins.services.connection.ConnectionSettingsProvider;
+import com.hpe.adm.octane.ideplugins.services.connection.HttpClientProvider;
 import com.hpe.adm.octane.ideplugins.services.connection.OctaneProvider;
 import com.hpe.adm.octane.ideplugins.services.di.ServiceModule;
+import com.hpe.adm.octane.ideplugins.services.util.ClientType;
+import com.sun.org.apache.xpath.internal.SourceTree;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
 import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import sun.net.www.http.HttpClient;
+
+import java.util.Collection;
 
 /**
  * Enables the use of the {@link Inject} annotation
  */
-public abstract class IntegrationTestBase {
+@WorkSpace(clean = true)
+public class IntegrationTestBase {
```

Comment:
The idea of this class is, something that all our other integration tests extend in order to get some util functionalty, to make testing easier. 

How I want the workspace annotation to be used:
It can be added optionally to any subclass of IntegrationTestBase.
Because the test class extends the base, it means that the inegration test base class is constructed before the class that extends it. In the constructor of the base, you will need to use reflection to check if the annotation is present. Based on this annotation, you can decide whether to construct a new workspace or provide an existing one. The workspace would automatically set in the inheritable connection settings field.
This is going to change later